[Bug]: MixCoord panic with FloatVector as ClusteringKey, DataNode round-robin panic with JSON as ClusteringKey after executing compact

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.6.9
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): wp
- SDK version(e.g. pymilvus v2.0.0rc2):go SDK v2.6.1
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: x86
- GPU: 
- Others:
```

### Current Behavior

When a collection schema is created with FieldTypeFloatVector or FieldTypeJSON field marked as ClusteringKey, no error occurs during collection creation, but executing the compact operation on the collection triggers severe node panics:

- When FieldTypeFloatVector is set as ClusteringKey: MixCoord node panics and crashes after compact is executed.
- When FieldTypeJSON is set as ClusteringKey: DataNode nodes panic in a round-robin manner (one after another) after compact is executed.

### Expected Behavior

Milvus should validate the schema during creation and reject FieldTypeFloatVector/FieldTypeJSON as ClusteringKey with a clear error message (since these types are not supported for ClusteringKey).
If schema creation is allowed, compact operation should execute normally without node panics.

`schema1 := entity.NewSchema().
        WithDynamicFieldEnabled(true).
        WithField(entity.NewField().WithName("key").WithDataType(entity.FieldTypeVarChar).WithIsPrimaryKey(true).WithMaxLength(1024)).
        WithField(entity.NewField().WithName("data").WithDataType(entity.FieldTypeFloatVector).WithDim(dim)).
        WithField(entity.NewField().WithName("metadata").WithDataType(entity.FieldTypeJSON).WithIsClusteringKey(true)).
        WithField(entity.NewField().WithName("shard").WithDataType(entity.FieldTypeInt64))


schema2 := entity.NewSchema().
        WithDynamicFieldEnabled(true).
        WithField(entity.NewField().WithName("key").WithDataType(entity.FieldTypeVarChar).WithIsPrimaryKey(true).WithMaxLength(1024)).
        WithField(entity.NewField().WithName("data").WithDataType(entity.FieldTypeFloatVector).WithDim(dim).WithIsClusteringKey(true)).
        WithField(entity.NewField().WithName("metadata").WithDataType(entity.FieldTypeJSON)).
        WithField(entity.NewField().WithName("shard").WithDataType(entity.FieldTypeInt64))
`

### Steps To Reproduce

```markdown
1.Use Milvus Go SDK to create a collection with either FieldTypeFloatVector or FieldTypeJSON as ClusteringKey.
2.Insert data into the collection.
3.Execute the compact operation on the collection.
```

### Milvus Log

[2026/02/02 01:42:59.448 +00:00] [DEBUG] [meta/target_manager.go:183] ["finish to update next targets for collection"] [collectionID=463910722792412534] [PartitionIDs="[463910722792412535]"]
[2026/02/02 01:42:59.448 +00:00] [DEBUG] [meta/target_manager.go:127] ["finish to update current target for collection"] [collectionID=463910722743513058] [segments="[463910722801739125,463910722802069125,463910722801519124,463910722801849125,463910722801959125,463910722801409124,463910722801629124]"] [channels="[by-dev-rootcoord-dml_0_463910722743513058v0]"] [version=1769996569449288846] [partStatsVersion="partitionStats:by-dev-rootcoord-dml_0_463910722743513058v0:[],"]
github.com/milvus-io/milvus/internal/datacoord.(*clusteringCompactionTask).doAnalyze
	/workspace/source/internal/datacoord/compaction_task_clustering.go:747 pc=0x5fc8c15


panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x5fc8c15]

goroutine 775 gp=0xc001971500 m=4 mp=0xc0000c1808 [running]:
panic({0x70981a0?, 0xb6e6980?})
	/usr/local/go/src/runtime/panic.go:811 +0x168 fp=0xc0015616b8 sp=0xc001561608 pc=0x28bfb28
github.com/milvus-io/milvus/pkg/v2/util/conc.(*poolOption).antsOptions.func1({0x70981a0, 0xb6e6980})
	/workspace/source/pkg/util/conc/options.go:56 +0x146 fp=0xc001561780 sp=0xc0015616b8 pc=0x52361c6
github.com/panjf2000/ants/v2.(*goWorker).run.func1.1()
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/worker.go:60 +0xd0 fp=0xc001561808 sp=0xc001561780 pc=0x5232bf0
panic({0x70981a0?, 0xb6e6980?})
	/usr/local/go/src/runtime/panic.go:792 +0x132 fp=0xc0015618b8 sp=0xc001561808 pc=0x28bfaf2
github.com/milvus-io/milvus/pkg/v2/util/conc.(*Pool[...]).Submit.func1.1()
	/workspace/source/pkg/util/conc/pool.go:75 +0x8a fp=0xc001561918 sp=0xc0015618b8 pc=0x59653ea
panic({0x70981a0?, 0xb6e6980?})
	/usr/local/go/src/runtime/panic.go:792 +0x132 fp=0xc0015619c8 sp=0xc001561918 pc=0x28bfaf2
runtime.panicmem(...)
	/usr/local/go/src/runtime/panic.go:262
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:925 +0x359 fp=0xc001561a28 sp=0xc0015619c8 pc=0x28c2dd9
github.com/milvus-io/milvus/internal/datacoord.(*clusteringCompactionTask).doAnalyze(0xc001034150)
	/workspace/source/internal/datacoord/compaction_task_clustering.go:747 +0x795 fp=0xc001561cc0 sp=0xc001561a28 pc=0x5fc8c15
github.com/milvus-io/milvus/internal/datacoord.(*clusteringCompactionTask).CreateTaskOnWorker(0xc001034150, 0x5c, {0x842daf8, 0xc0016d3af0})
	/workspace/source/internal/datacoord/compaction_task_clustering.go:125 +0x554 fp=0xc001561e78 sp=0xc001561cc0 pc=0x5fb8934
github.com/milvus-io/milvus/internal/datacoord/task.(*globalTaskScheduler).schedule.func1()
	/workspace/source/internal/datacoord/task/global_scheduler.go:183 +0x151 fp=0xc001561f20 sp=0xc001561e78 pc=0x5e53191
github.com/milvus-io/milvus/pkg/v2/util/conc.(*Pool[...]).Submit.func1()
	/workspace/source/pkg/util/conc/pool.go:82 +0x9e fp=0xc001561f80 sp=0xc001561f20 pc=0x59652de
github.com/panjf2000/ants/v2.(*goWorker).run.func1()
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/worker.go:73 +0x8a fp=0xc001561fe0 sp=0xc001561f80 pc=0x5232aaa
runtime.goexit({})
	/usr/local/go/src/runtime/asm_amd64.s:1700 +0x1 fp=0xc001561fe8 sp=0xc001561fe0 pc=0x28c9561
created by github.com/panjf2000/ants/v2.(*goWorker).run in goroutine 615
	/go/pkg/mod/github.com/panjf2000/ants/v2@v2.11.3/worker.go:50 +0x65

[coorderror.txt](https://github.com/user-attachments/files/25061749/coorderror.txt)

### Anything else?

_No response_

--- a/internal/datacoord/compaction_inspector.go
+++ b/internal/datacoord/compaction_inspector.go
@@ -178,19 +178,20 @@ func (c *compactionInspector) getCompactionTasksNumBySignalID(triggerID int64) i
 }
 
 func newCompactionInspector(meta CompactionMeta,
-	allocator allocator.Allocator, handler Handler, scheduler task.GlobalScheduler, ievm IndexEngineVersionManager,
+	allocator allocator.Allocator, handler Handler, scheduler task.GlobalScheduler, analyzeScheduler task.GlobalScheduler, ievm IndexEngineVersionManager,
 ) *compactionInspector {
 	capacity := paramtable.Get().DataCoordCfg.CompactionTaskQueueCapacity.GetAsInt()
 	return &compactionInspector{
-		queueTasks:     NewCompactionQueue(capacity, getPrioritizer()),
-		meta:           meta,
-		allocator:      allocator,
-		stopCh:         make(chan struct{}),
-		executingTasks: make(map[int64]CompactionTask),
-		cleaningTasks:  make(map[int64]CompactionTask),
-		handler:        handler,
-		scheduler:      scheduler,
-		ievm:           ievm,
+		queueTasks:       NewCompactionQueue(capacity, getPrioritizer()),
+		meta:             meta,
+		allocator:        allocator,
+		stopCh:           make(chan struct{}),
+		executingTasks:   make(map[int64]CompactionTask),
+		cleaningTasks:    make(map[int64]CompactionTask),
+		handler:          handler,
+		scheduler:        scheduler,
+		analyzeScheduler: analyzeScheduler,
+		ievm:             ievm,
 	}
 }
 

--- a/internal/datacoord/compaction_inspector_test.go
+++ b/internal/datacoord/compaction_inspector_test.go
@@ -58,7 +58,7 @@ func (s *CompactionPlanHandlerSuite) SetupTest() {
 	s.mockMeta.EXPECT().SaveCompactionTask(mock.Anything, mock.Anything).Return(nil).Maybe()
 	s.mockAlloc = allocator.NewMockAllocator(s.T())
 	mockScheduler := task.NewMockGlobalScheduler(s.T())
-	s.handler = newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, newMockVersionManager())
+	s.handler = newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, mockScheduler, newMockVersionManager())
 	s.mockHandler = NewNMockHandler(s.T())
 	s.mockHandler.EXPECT().GetCollection(mock.Anything, mock.Anything).Return(&collectionInfo{}, nil).Maybe()
 }
@@ -461,7 +461,7 @@ func (s *CompactionPlanHandlerSuite) TestCompactionQueueFull() {
 			t.QueryTaskOnWorker(cluster)
 		}
 	}).Maybe()
-	s.handler = newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, newMockVersionManager())
+	s.handler = newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, mockScheduler, newMockVersionManager())
 
 	t1 := newMixCompactionTask(&datapb.CompactionTask{
 		TriggerID: 1,
@@ -495,7 +495,7 @@ func (s *CompactionPlanHandlerSuite) TestExecCompactionPlan() {
 			t.QueryTaskOnWorker(cluster)
 		}
 	}).Maybe()
-	handler := newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, newMockVersionManager())
+	handler := newCompactionInspector(s.mockMeta, s.mockAlloc, nil, mockScheduler, mockScheduler, newMockVersionManager())
 
 	task := &datapb.CompactionTask{
 		TriggerID: 1,

--- a/internal/datacoord/server.go
+++ b/internal/datacoord/server.go
@@ -760,7 +760,7 @@ func (s *Server) initExternalCollectionInspector() {
 }
 
 func (s *Server) initCompaction() {
-	cph := newCompactionInspector(s.meta, s.allocator, s.handler, s.globalScheduler, s.indexEngineVersionManager)
+	cph := newCompactionInspector(s.meta, s.allocator, s.handler, s.globalScheduler, s.globalScheduler, s.indexEngineVersionManager)
 	cph.loadMeta()
 	s.compactionInspector = cph
 	s.compactionTriggerManager = NewCompactionTriggerManager(s.allocator, s.handler, s.compactionInspector, s.meta, s.indexEngineVersionManager)

--- a/internal/datacoord/server_test.go
+++ b/internal/datacoord/server_test.go
@@ -1652,7 +1652,7 @@ func TestGetCompactionState(t *testing.T) {
 				{State: datapb.CompactionTaskState_timeout},
 				{State: datapb.CompactionTaskState_timeout},
 			})
-		mockHandler := newCompactionInspector(mockMeta, nil, nil, nil, newMockVersionManager())
+		mockHandler := newCompactionInspector(mockMeta, nil, nil, nil, nil, newMockVersionManager())
 		svr.compactionInspector = mockHandler
 		resp, err := svr.GetCompactionState(context.Background(), &milvuspb.GetCompactionStateRequest{CompactionID: 1})
 		assert.NoError(t, err)

--- a/internal/proxy/task.go
+++ b/internal/proxy/task.go
@@ -327,6 +327,10 @@ func (t *createCollectionTask) validateClusteringKey(ctx context.Context) error 
 	idx := -1
 	for i, field := range t.schema.Fields {
 		if field.GetIsClusteringKey() {
+			if !typeutil.IsClusteringKeyType(field.GetDataType()) {
+				return merr.WrapErrCollectionIllegalSchema(t.CollectionName,
+					fmt.Sprintf("clustering key field %s has unsupported data type %s", field.Name, field.GetDataType().String()))
+			}
 			if typeutil.IsVectorType(field.GetDataType()) &&
 				!paramtable.Get().CommonCfg.EnableVectorClusteringKey.GetAsBool() {
 				return merr.WrapErrCollectionVectorClusteringKeyNotAllowed(t.CollectionName)

--- a/internal/proxy/task_test.go
+++ b/internal/proxy/task_test.go
@@ -4841,6 +4841,145 @@ func TestClusteringKey(t *testing.T) {
 		err = createCollectionTask.PreExecute(ctx)
 		assert.Error(t, err)
 	})
+
+	t.Run("create collection with json clustering key not supported", func(t *testing.T) {
+		fieldName2Type := make(map[string]schemapb.DataType)
+		fieldName2Type["int64_field"] = schemapb.DataType_Int64
+		fieldName2Type["varChar_field"] = schemapb.DataType_VarChar
+		schema := constructCollectionSchemaByDataType(collectionName, fieldName2Type, "int64_field", false)
+		jsonField := &schemapb.FieldSchema{
+			Name:            "json_field",
+			DataType:        schemapb.DataType_JSON,
+			IsClusteringKey: true,
+		}
+		schema.Fields = append(schema.Fields, jsonField)
+		vecField := &schemapb.FieldSchema{
+			Name:     "fvec_field",
+			DataType: schemapb.DataType_FloatVector,
+			TypeParams: []*commonpb.KeyValuePair{
+				{
+					Key:   common.DimKey,
+					Value: strconv.Itoa(testVecDim),
+				},
+			},
+		}
+		schema.Fields = append(schema.Fields, vecField)
+		marshaledSchema, err := proto.Marshal(schema)
+		assert.NoError(t, err)
+
+		createCollectionTask := &createCollectionTask{
+			Condition: NewTaskCondition(ctx),
+			CreateCollectionRequest: &milvuspb.CreateCollectionRequest{
+				Base: &commonpb.MsgBase{
+					MsgID:     UniqueID(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+					Timestamp: Timestamp(time.Now().UnixNano()),
+				},
+				DbName:         "",
+				CollectionName: collectionName,
+				Schema:         marshaledSchema,
+				ShardsNum:      shardsNum,
+			},
+			ctx:      ctx,
+			mixCoord: qc,
+			result:   nil,
+			schema:   nil,
+		}
+		err = createCollectionTask.PreExecute(ctx)
+		assert.Error(t, err)
+	})
+
+	t.Run("create collection with bool clustering key not supported", func(t *testing.T) {
+		fieldName2Type := make(map[string]schemapb.DataType)
+		fieldName2Type["int64_field"] = schemapb.DataType_Int64
+		fieldName2Type["varChar_field"] = schemapb.DataType_VarChar
+		schema := constructCollectionSchemaByDataType(collectionName, fieldName2Type, "int64_field", false)
+		boolField := &schemapb.FieldSchema{
+			Name:            "bool_field",
+			DataType:        schemapb.DataType_Bool,
+			IsClusteringKey: true,
+		}
+		schema.Fields = append(schema.Fields, boolField)
+		vecField := &schemapb.FieldSchema{
+			Name:     "fvec_field",
+			DataType: schemapb.DataType_FloatVector,
+			TypeParams: []*commonpb.KeyValuePair{
+				{
+					Key:   common.DimKey,
+					Value: strconv.Itoa(testVecDim),
+				},
+			},
+		}
+		schema.Fields = append(schema.Fields, vecField)
+		marshaledSchema, err := proto.Marshal(schema)
+		assert.NoError(t, err)
+
+		createCollectionTask := &createCollectionTask{
+			Condition: NewTaskCondition(ctx),
+			CreateCollectionRequest: &milvuspb.CreateCollectionRequest{
+				Base: &commonpb.MsgBase{
+					MsgID:     UniqueID(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+					Timestamp: Timestamp(time.Now().UnixNano()),
+				},
+				DbName:         "",
+				CollectionName: collectionName,
+				Schema:         marshaledSchema,
+				ShardsNum:      shardsNum,
+			},
+			ctx:      ctx,
+			mixCoord: qc,
+			result:   nil,
+			schema:   nil,
+		}
+		err = createCollectionTask.PreExecute(ctx)
+		assert.Error(t, err)
+	})
+
+	t.Run("create collection with array clustering key not supported", func(t *testing.T) {
+		fieldName2Type := make(map[string]schemapb.DataType)
+		fieldName2Type["int64_field"] = schemapb.DataType_Int64
+		fieldName2Type["varChar_field"] = schemapb.DataType_VarChar
+		schema := constructCollectionSchemaByDataType(collectionName, fieldName2Type, "int64_field", false)
+		arrayField := &schemapb.FieldSchema{
+			Name:            "array_field",
+			DataType:        schemapb.DataType_Array,
+			IsClusteringKey: true,
+			ElementType:     schemapb.DataType_Int64,
+		}
+		schema.Fields = append(schema.Fields, arrayField)
+		vecField := &schemapb.FieldSchema{
+			Name:     "fvec_field",
+			DataType: schemapb.DataType_FloatVector,
+			TypeParams: []*commonpb.KeyValuePair{
+				{
+					Key:   common.DimKey,
+					Value: strconv.Itoa(testVecDim),
+				},
+			},
+		}
+		schema.Fields = append(schema.Fields, vecField)
+		marshaledSchema, err := proto.Marshal(schema)
+		assert.NoError(t, err)
+
+		createCollectionTask := &createCollectionTask{
+			Condition: NewTaskCondition(ctx),
+			CreateCollectionRequest: &milvuspb.CreateCollectionRequest{
+				Base: &commonpb.MsgBase{
+					MsgID:     UniqueID(uniquegenerator.GetUniqueIntGeneratorIns().GetInt()),
+					Timestamp: Timestamp(time.Now().UnixNano()),
+				},
+				DbName:         "",
+				CollectionName: collectionName,
+				Schema:         marshaledSchema,
+				ShardsNum:      shardsNum,
+			},
+			ctx:      ctx,
+			mixCoord: qc,
+			result:   nil,
+			schema:   nil,
+		}
+		err = createCollectionTask.PreExecute(ctx)
+		assert.Error(t, err)
+	})
 }
 
 func TestAlterCollectionCheckLoaded(t *testing.T) {

--- a/pkg/util/typeutil/schema.go
+++ b/pkg/util/typeutil/schema.go
@@ -635,6 +635,20 @@ func IsVectorArrayType(dataType schemapb.DataType) bool {
 	return dataType == schemapb.DataType_ArrayOfVector
 }
 
+// IsClusteringKeyType returns true if the data type is supported as a clustering key.
+// Supported scalar types: Int8, Int16, Int32, Int64, Float, Double, VarChar, String, FloatVector.
+func IsClusteringKeyType(dataType schemapb.DataType) bool {
+	switch dataType {
+	case schemapb.DataType_Int8, schemapb.DataType_Int16,
+		schemapb.DataType_Int32, schemapb.DataType_Int64,
+		schemapb.DataType_Float, schemapb.DataType_Double,
+		schemapb.DataType_VarChar, schemapb.DataType_String:
+		return true
+	default:
+		return dataType == schemapb.DataType_FloatVector
+	}
+}
+
 // IsIntegerType returns true if input is an integer type, otherwise false
 func IsIntegerType(dataType schemapb.DataType) bool {
 	switch dataType {

--- a/pkg/util/typeutil/schema_test.go
+++ b/pkg/util/typeutil/schema_test.go
@@ -5549,3 +5549,29 @@ func TestUpdateFieldDataByColumn(t *testing.T) {
 		assert.Equal(t, [][]byte{{10}, {2}, {20}}, base.GetVectors().GetSparseFloatVector().Contents)
 	})
 }
+
+func TestIsClusteringKeyType(t *testing.T) {
+	// Supported scalar types
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Int8))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Int16))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Int32))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Int64))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Float))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_Double))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_VarChar))
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_String))
+
+	// Supported vector types
+	assert.True(t, IsClusteringKeyType(schemapb.DataType_FloatVector))
+
+	// Unsupported types
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_JSON))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Bool))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Array))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Geometry))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Text))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Timestamptz))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_BinaryVector))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_Float16Vector))
+	assert.False(t, IsClusteringKeyType(schemapb.DataType_BFloat16Vector))
+}

